File: navigation/src/main/java/com/example/navigation/stepsEngine/field/PaymentStatusField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.payment.FlowState;

public class PaymentStatusField extends Field {

    @Override
    public String get(FlowState flowState) {
        return flowState.getPaymentStatus().toUpperCase();
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/PaymentStatusField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState
import kotlinx.android.parcel.Parcelize

@Parcelize
class PaymentStatusField : Field() {
    override operator fun get(flowState: FlowState): String? {
        return flowState.paymentStatus?.toUpperCase()
    }

    override fun getId(): String {
        return FieldName.PAYMENT_STATUS
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}